File: MasterDirectory/MasterDirectory.Web/Modules/MarketIntelligence/CategoriaMarketIntelligence/CategoriaMarketIntelligenceForm.cs
using Serenity.ComponentModel;
using System;

namespace MasterDirectory.MarketIntelligence.Forms;

[FormScript("MarketIntelligence.CategoriaMarketIntelligence")]
[BasedOnRow(typeof(CategoriaMarketIntelligenceRow), CheckNames = true)]
public class CategoriaMarketIntelligenceForm
{
    [Tab("General")]
    public string LocalSap { get; set; }
    public string Nse { get; set; }
    public string Cluster { get; set; }
    public string Brick { get; set; }
    //public DateTime DtRegistro { get; set; }
}

File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/MarketIntelligence/CategoriaMarketIntelligenceForm.ts
import { StringEditor, LookupEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CategoriaMarketIntelligenceForm {
    LocalSap: StringEditor;
    Nse: LookupEditor;
    Cluster: LookupEditor;
    Brick: StringEditor;
}

export class CategoriaMarketIntelligenceForm extends PrefixedContext {
    static formKey = 'MarketIntelligence.CategoriaMarketIntelligence';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoriaMarketIntelligenceForm.init)  {
            CategoriaMarketIntelligenceForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;

            initFormType(CategoriaMarketIntelligenceForm, [
                'LocalSap', w0,
                'Nse', w1,
                'Cluster', w1,
                'Brick', w0
            ]);
        }
    }
}